src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -2221,13 +2222,19 @@ bool CanDeleteSecretKey()
                         break;
                     case EtlType.GenAi:
                     {
-                        var aiIntegration = JsonDeserializationCluster.GenAiConfiguration(etlConfiguration);
-                        aiIntegration.Validate(out var aiIntegrationErr, validateName: false, validateConnection: false);
-                        if (ValidateConnectionString(rawRecord, aiIntegration.ConnectionStringName, aiIntegration.EtlType) == false)
-                            aiIntegrationErr.Add(
-                                $"Could not find connection string named '{aiIntegration.ConnectionStringName}'. Please supply an existing connection string.");
-                        ThrowInvalidConfigurationIfNecessary(etlConfiguration, aiIntegrationErr);
-                        command = new AddGenAiCommand(aiIntegration, databaseName, raftRequestId);
+                        if (rawRecord.IsSharded) 
+                            throw new NotSupportedInShardingException("GenAI is currently not supported in sharding");
+
+                        var genAi = JsonDeserializationCluster.GenAiConfiguration(etlConfiguration);
+                        genAi.Validate(out var genAiErr, validateName: false, validateConnection: false);
+                        if (ValidateConnectionString(rawRecord, genAi.ConnectionStringName, genAi.EtlType) == false)
+                            genAiErr.Add($"Could not find connection string named '{genAi.ConnectionStringName}'. Please supply an existing connection string.");
+                        ThrowInvalidConfigurationIfNecessary(etlConfiguration, genAiErr);
+
+                        if (genAi.ProcessNewDocumentsOnly)
+                            await UpdateGenAiStateToLastEtag(databaseName, genAi);
```

Comment:
This is where it is used, and that is a great example on why it should _not_ be on the configuration.
It should be at the parameters for this command.